arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/RaiseRacing.kt

```diff
@@ -0,0 +1,202 @@
+package arrow.fx.coroutines
+
+import arrow.core.raise.Raise
+import arrow.core.raise.recover
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitCancellation
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.selects.SelectBuilder
+import kotlinx.coroutines.selects.select
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.time.Duration
+
+/**
+ * A scope that combines [CoroutineScope] and [Raise] capabilities, allowing for both coroutine operations
+ * and error handling within a single scope.
+ *
+ * @param Error The type of error that can be raised within this scope.
+ */
+public interface RaiseScope<in Error> : CoroutineScope, Raise<Error>
+
+/**
+ * A scope that allows racing multiple coroutine blocks against each other with [Raise] context receiver support.
+ * The first block to complete successfully will provide the result, and all other blocks will be cancelled.
+ *
+ * @param Error The error type that can be raised by the racing blocks.
+ * @param Result The type of value that will be returned by the racing blocks.
+ */
+public interface RaiseRacingScope<in Error, in Result> {
```

Comment:
Shouldn't this extend `RacingScope`?

Reply:
It can't because the functions won't resolve nicely without requiring different names.